src/d3a/models/market/one_sided.py

```diff
@@ -173,12 +174,12 @@ def determine_offer_price(self, energy_portion, energy, trade_rate,
                 energy, trade_rate, energy_portion, orig_offer_price
             )
         else:
-            revenue, fees, trade_rate_incl_fees = \
+            revenue, grid_fee_rate, trade_rate_incl_fees = \
```

Comment:
Looking at the diff below, I think it would be much simpler and more consistent if the `determine_offer_price` function returned grid_fee_price instead of grid_fee_rate, since you need only the fee price after this function returns.

Reply:
The reason I did it this way is to take into account the both cases, i.e. accepting full as well as partial offers. We can discuss offline too if this still doesn't seems like a good approach to you.

Reply:
Since you have the energy_portion here, you can calculate the fee price for both cases. The reason why I am insisting in this is because the same function in the balancing market returns fees as price (if I understand the PR correctly), and not as rate. Also, having a price function here would allow us to only use energy rates in the context of this function, and once the `determine_offer_price` is called the market is only going to use prices as a unit.

Reply:
Okay, I have adapted this comment, now `determine_offer_price` will return grid_fee in price.